firebase-installations/src/main/java/com/google/firebase/installations/FirebaseInstallations.java

```diff
@@ -169,6 +180,15 @@ private static boolean persistedFidMissingOrInErrorState(PersistedFidEntry persi
     };
   }
 
+  @NonNull
+  private <F, T> Continuation<F, T> call(@NonNull Supplier<T> supplier) {
+    return t -> {
+      // Waiting for Task that registers FID on the FIS Servers
+      latch.await(10, TimeUnit.SECONDS);
+      return supplier.get();
+    };
+  }
+
   private PersistedFidEntry createAndPersistNewFid() throws FirebaseInstallationsException {
```

Comment:
add javadoc or comment? same for the others

Reply:
Done. PTAL.

Reply:
Done. Thanks.